src/classes/ERR_Handler.cls

```diff
@@ -138,62 +138,155 @@ public class ERR_Handler {
     	}
     	return errors;
     }
-        
+
+	/*******************************************************************************************************
+    * @description Processes the results of a group DML operations caused by a batch, future, or queued job and performed
+    * 				using the Database class to determine if errors occurred. It does not store errors or send error email
+    * 				notifications. These operations should be performed after the rollback.
+    *
+    * @param dmlResults The results of a DML batch operation performed using the Database class.
+    * @param sObjects The records the DML operation was performed on.
+    * @return Errors The errors with a single list that occurred during a batch DML operation performed using the Database class.
+    */
+	public static Errors getJobErrors(List<Object> dmlResults, List<SObject> sObjects) {
+		Errors errors = new Errors();
+		List<String> errorsFound = new List<String>();
+
+		if(dmlResults != null) {
+			for(Integer i=0; i<dmlResults.size(); i++) {
+
+				Boolean isSuccess = isDmlSuccess(dmlResults[i]);
+
+				if(!isSuccess) {
+					List<Database.Error> errList = getDmlResultErrors(dmlResults[i]);
+					Integer totalErrors = 0;
+					String errDetailsMessage = '';
+
+					for(Database.Error err : errList) {
+						totalErrors++;
+						errDetailsMessage += err.getStatusCode() + ': ' + ERR_ExceptionHandler.beautifyExceptionMessage(err.getMessage());
+
+						if(errDetailsMessage.endsWith('.')){
+							errDetailsMessage += ' ';
+						} else if(!errDetailsMessage.endsWith('. ')){
+							errDetailsMessage += '. ';
+						}
+					}
+
+					String fullObjectErrMessage = 'There were ' + totalErrors + ' errors saving to object ' + sObjects[i].Id + '. ' + errDetailsMessage;
+
+					errorsFound.add(fullObjectErrMessage);
+				}
+			}
+			if(errorsFound.size() > 0){
+				String errorMessage = String.join(errorsFound, '\n* ');
+				errors.errorsExist = true;
+
+				Error__c err = createJobError(errorMessage);
+				errors.errorRecords.add(err);
+
+				//We cannot use addError on the record here, because if we do, the whole transaction will be rolled back, and
+				//no error record will be saved, or error notification sent.
+			}
+		}
+		return errors;
+	}
+
+	/*******************************************************************************************************
+    * @description Checks if a given DML operation has succeeded depending on the type of DML result.
+    * @param dmlResults
+    * @return isSuccess Returns true if all DML operations passed successful; returns false if there was a DML exception.
+    */
+	public static boolean isDmlSuccess(Object dmlResults) {
```

Comment:
Probably a bit nit-picky, but since this refers to a single instance of SaveResult, etc., should the var name here be `dmlResult` instead of `dmlResults`?

Reply:
Also, do you think this method could be rewritten as the following? Basically, both methods take a similar action on the `dmlResult` parameter. If `getDmlResultErrors` returns 0, then I would expect that isSuccess is true (unless maybe there's a scenario where isSuccess is false and there are no errors).
`return getDmlResultErrors(dmlResult).size() == 0;`

Reply:
Changed var name to a singular. And agreed, I had made a note to condense those and that got list in the mix somewhere. As far as I'm aware, there shouldn't be a scenario where there's a failure without any messaging!